File: apps/tauri/KawaCAD/src/features/inspector/domain/inspectorValueFormatting.ts
type Value = Record<string, number | string> | undefined;
type Parameter = { id: string; valueMm: number };

export function formatInspectorNumber(value: number | undefined): string {
  return typeof value === "number" && Number.isFinite(value) ? value.toFixed(2) : "";
}

export function resolveInspectorValue(
  value: Value,
  parameters: readonly Parameter[],
  fixedKey: "fixedMm" | "fixedDegrees" = "fixedMm",
): number | undefined {
  const fixedValue = value?.[fixedKey];
  if (typeof fixedValue === "number" && Number.isFinite(fixedValue)) return fixedValue;
  const parameterID = value?.parameter;
  if (typeof parameterID !== "string") return undefined;
  return parameters.find((parameter) => parameter.id === parameterID)?.valueMm;
}

File: apps/macos/KawaCAD/Sources/KawaCADApp/Features/Inspector/Domain/InspectorValueFormatting.swift
import Foundation

enum InspectorValueFormatting {
  static func format(_ value: Double?) -> String {
    guard let value, value.isFinite else { return "" }
    return String(format: "%.2f", value)
  }

  static func resolvedText(
    fixedValue: Double?,
    parameterID: String?,
    parameters: [ProjectParameter]
  ) -> String {
    if let fixedValue {
      return format(fixedValue)
    }
    guard let parameterID,
      let parameter = parameters.first(where: { $0.id == parameterID })
    else {
      return ""
    }
    return format(parameter.valueMM)
  }
}
